mx.c

```diff
@@ -1808,6 +1808,12 @@ enum MxStatus mx_mbox_check_stats(struct Mailbox *m, uint8_t flags)
     struct EventMailbox ev_m = { m };
     notify_send(m->notify, NT_MAILBOX, NT_MAILBOX_CHANGE, &ev_m);
   }
+  if (rc == MX_STATUS_NEW_MAIL && !m->notified)
+  {
+    struct EventMailbox ev_m = { m };
+    notify_send(m->notify, NT_MAILBOX, NT_MAILBOX_NEW_MAIL, &ev_m);
+    m->notified = true;
+  }
```

Comment:
We want this to be a step up from the old feature...

I suggest a simple mechanism to start:
- Create a `static time_t` in this file that stores the time of the last check
- The first time through this code, `(t == 0)`, we update the time to **now**, but don't send a notification
- Next time, we search the `Mailbox` for any `Email`s whose `received` time is after `t` and put them in an array
- Send a notification with the `Mailbox` and the `EmailArray`

Reply:
When you say `EmailArray` do you mean an existing type? I do see `MdEmailArray` but that seems to be something different. Should I define a similar one for the purpose of the notifications?

Reply:
> `EmailArray` do you mean an existing type?

No, it'll be a new type using `ARRAY` (which works like a C++ template).
We want an `ARRAY` of `struct Email *` (similar to `MdEmailArray`).
As it will be general-purpose, we can put it at the end of the `struct Email` definition.

(Note: `ARRAY` supports creating arrays of **objects**, e.g. `AliasViewArray`)

We'll use the `ARRAY` as a temporary collection.
Since the `Email`s will be owned by the `Mailbox` we only need to `ARRAY_FREE()` afterwards.

We already do something similar in the Index, `op_pipe()`.
Unfortunately, that was written before we'd invented `ARRAY`.

https://github.com/neomutt/neomutt/blob/ef0cb550ecc8bc4e918c2d8efc16cad5dc39c2c6/index/functions.c#L1763-L1766

Reply:
Got it, thanks for clarifying.